### ISSUE TITLE
Data persists in DB after changing conditional field

### ISSUE THREAD

**Issue by creative-andrew** [payloadcms/payload#3147]
### Link to reproduction

https://github.com/payloadcms/public-demo/pull/25/files

### To Reproduce

We should not store unwanted information in the collection after unchecking the condition fields.

Please check the following PR recently merged in the payload demo website that exemplifies this problem: https://github.com/payloadcms/public-demo/pull/25

An easy example to illustrate the problem can be accomplished with the following steps. 

1. Create any conditional field. For example, a `Show image?` checkbox field that makes visible a field called `media`.
2. Add an image to the media `field`.
3. Make a call to the API to verify that the data is coming correctly.
4. Uncheck `Show image? `field to make the media field disappear.
5. Make a call to the API and you will notice that object with data of the media field is still present in the DB/api response. 

### Describe the Bug

We should not store unwanted information in the collection when the condition is not met.

It would be great to pass a configuration option wherever the condition is defined to enable this behavior. 

Right now I am forced to do the following every time I define a condition in a component:

```
   admin: {
          condition: (_, siblingData) => siblingData?.type === 'reference',
        },
        hooks: {
          beforeChange: [
            ({ siblingData, value }) => (siblingData?.type !== "reference" ? null : value),
          ],
        },
```

It would be great to have this out of the box, by doing something like:

```
   admin: {
          condition: (_, siblingData) => siblingData?.type === 'reference',
        },
         removeValuesOnCondition: true
```
So whenever the condition is not true, it should not stored (reset) to the default empty value. 

### Payload Version

1.9.0

**Comment by DanRibbens** [comment]
@tylandavis, you're right that we wouldn't want to run the condition on the backend as an extra step to then determine if a field should be unset or not.

I think the behavior needs to be baked into the field without adding additional config properties. If a user is editing a document on the admin panel and changes a conditional field so that another field that had a previous value is removed, then on save the property needs to be assigned `null` in the PATCH so that the backend can handle it without the need for a beforeChange hook or anything else.

Would that satisfy your concerns @creative-andrew?

**Comment by creative-andrew** [comment]
That would be great and I think it should be the expected result. Namely, we shouldn't store extra data when a condition is not met and the user saves the document.
